sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/JsonSerializable.java

```diff
@@ -252,29 +243,61 @@ public void remove(String propertyName) {
      * @param value the value of the property.
      */
     @SuppressWarnings({"unchecked", "rawtypes"})
-    public <T> void set(String propertyName, T value) {
+    public <T> void set(String propertyName, T value, CosmosItemSerializer itemSerializer) {
```

Comment:
We found this is a breaking change when upgrade azure-cosmos from 4.58 to 4.59 since it requires an extra parameter, though it's always `CosmosItemSerializer.DEFAULT_SERIALIZER`. 
Is it a missing in release note to call out this API breaking change?

Reply:
It is not a breaking change - because this is an internal API. The implementation package is not exported and is not part of public surface are. Breaking changes in the implementation package can and will happen rather frequently - even in hotfix versions - and it is simply a bug to take a direct dependency on it - modules are not enforced in Java 8 technically - but more recent Java versions would even prevent your app to use these APIs.